File: 프로그래머스/5/214292. 재밌는 레이싱 경기장 설계하기/재밌는 레이싱 경기장 설계하기.js
function solution(heights) {
  heights.sort((a, b) => a - b);

  const n = heights.length;
  const minusV = [];

  if (n % 2 === 1) { // 홀수
    for (let i = 0; i < Math.floor(n / 2); i++) {
      minusV.push(heights[i + Math.floor(n / 2)] - heights[i]);
    }
    minusV.push(heights[n - 1] - heights[Math.floor(n / 2)]);
    minusV.sort((a, b) => a - b);
    return minusV[1];
  } else { // 짝수
    for (let i = 0; i < n / 2; i++) {
      minusV.push(heights[i + n / 2] - heights[i]);
    }
    minusV.sort((a, b) => a - b);
    return minusV[0];
  }
}

File: 프로그래머스/5/214292. 재밌는 레이싱 경기장 설계하기/재밌는 레이싱 경기장 설계하기.java
import java.util.Arrays;

class Solution {
    public int solution(int[] heights) {
        Arrays.sort(heights);
        int n = heights.length;

        int[] minusV = new int[n / 2 + 1];
        int idx = 0;

        if (n % 2 == 1) { // 홀수
            for (int i = 0; i < n / 2; i++) {
                minusV[idx++] = heights[i + n / 2] - heights[i];
            }
            minusV[idx++] = heights[n - 1] - heights[n / 2];

            Arrays.sort(minusV, 0, idx);
            return minusV[1];
        } else { // 짝수
            for (int i = 0; i < n / 2; i++) {
                minusV[idx++] = heights[i + n / 2] - heights[i];
            }

            Arrays.sort(minusV, 0, idx);
            return minusV[0];
        }
    }
}